src/MultiMessageReceiver.sol

```diff
@@ -143,7 +135,9 @@ contract MultiMessageReceiver is IMultiMessageReceiver, ExecutorAware, Initializ
 
         /// stores the message if the amb is the first one delivering the message
```

Comment:
suggest we validate that `message.target` is not `address(0)` in this function. It is a condition enforced in the sender, but might be good to ensure here as well.